src/main/java/ch/njol/skript/conditions/CondEntityIsIn.java

```diff
@@ -0,0 +1,138 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Is in Rain/Lava/Water/BubbleColumn")
+@Description("Checks whether an entity is in Rain/Lava/Water/BubbleColumn")
+@Examples({"player is in rain",
+		"player is in water",
+		"player is in lava",
+		"player is in bubble column",
+		"player is in water or rain",
+		"player is in water or bubble column",
+		"player is in water or rain or bubble column"})
+@RequiredPlugins("Minecraft 1.16+, Paper 1.16+ (rain, lava and bubble column)")
+@Since("INSERT VERSION")
+public class CondEntityIsIn extends PropertyCondition<Entity> {
+	
+	static {
+		StringBuilder patterns = new StringBuilder("");
+		if (Skript.methodExists(Entity.class, "isInWater"))
+			patterns.append("1¦water|");
+		if (Skript.methodExists(Entity.class, "isInLava"))
+			patterns.append("2¦lava|");
+		if (Skript.methodExists(Entity.class, "isInBubbleColumn"))
+			patterns.append("3¦bubble[ ]column|");
+		if (Skript.methodExists(Entity.class, "isInRain"))
+			patterns.append("4¦rain|");
+		if (Skript.methodExists(Entity.class, "isInWaterOrRain"))
+			patterns.append("5¦water or rain|");
+		if (Skript.methodExists(Entity.class, "isInWaterOrBubbleColumn"))
+			patterns.append("6¦water or bubble[ ]column|");
+		if (Skript.methodExists(Entity.class, "isInWaterOrRainOrBubbleColumn"))
+			patterns.append("7¦water or rain or bubble[ ]column");
+
+		if (patterns.toString().endsWith("|")) // Remove last empty '|' if exists
+			patterns.deleteCharAt(patterns.length() - 1);
+
+		register(CondEntityIsIn.class, PropertyType.BE, "in (" + patterns + ")", "entities");
+	}
+
+	static final int IN_WATER = 1;
+	static final int IN_LAVA = 2;
+	static final int IN_BUBBLE_COLUMN = 3;
+	static final int IN_RAIN = 4;
+	static final int IN_WATER_OR_RAIN = 5;
+	static final int IN_WATER_OR_BUBBLE_COLUMN = 6;
+	static final int IN_WATER_OR_RAIN_OR_BUBBLE_COLUMN = 7;
```

Comment:
```suggestion
	private static final int IN_WATER = 1, IN_LAVA = 2, IN_BUBBLE_COLUMN = 3, IN_RAIN = 4, IN_WATER_OR_RAIN = 5, 
			IN_WATER_OR_BUBBLE_COLUMN = 6, IN_WATER_OR_RAIN_OR_BUBBLE_COLUMN = 7;
```
I think this would work better :)